src/com/carolinarollergirls/scoreboard/defaults/DefaultScoreBoardModel.java

```diff
@@ -319,74 +313,58 @@ public void timeout(TeamModel team, boolean review) {
 		}
 	}
 
-	public void unStartJam() {
-		synchronized (runLock) {
-			if (!getClock(Clock.ID_JAM).isRunning())
-				return;
-
-			requestBatchStart();
-			if (lineupClockWasRunning)
-				getClockModel(Clock.ID_LINEUP).unstop();
-			if (timeoutClockWasRunning) {
-				setTimeoutOwner(lastTimeoutOwner);
-				setOfficialReview(wasOfficialReview);
-				getClockModel(Clock.ID_TIMEOUT).unstop();
+	protected void createSnapshot(String type) {
+		snapshot = new ScoreBoardSnapshot(this, type);
+	}
+	protected void restoreSnapshot() {
+		relapseTime = System.currentTimeMillis() - snapshot.getSnapshotTime();
```

Comment:
How are you unittesting this given that System.currentTimeMillis isn't stubbed out?

Reply:
I'm doing a plausibilty check on `relapseTime` (>0) and verify sure that this is the time that is being applied to the clocks. I didn't see a way to make sure that this is the actual elapsed wall lock time.

Reply:
https://github.com/prometheus/client_java/blob/master/simpleclient/src/main/java/io/prometheus/client/Gauge.java#L139 is one way to handle it.

Reply:
Wrapping System.currentTimeMillis() into a class? The idea then being that we stub  this class for tests?
That would work but it's a lot of overhead just to test two simple lines of code.

Reply:
Yes. That's the standard way to do it, every approach I've seen boils down to pretty much that.

Reply:
Turns out we already have a class that fits this description sufficiently close and is instrumented for unit tests: `UpdateClockTimerTask`. (It returns ms since startup instead of system time but as we only want to measure a difference, this is perfectly fine.)